tests/Core/ConfigTest.php

```diff
@@ -0,0 +1,204 @@
+     *
+     * @return void
+     */
+    public function testReadonlyConfigThrowsPhpCodeSnifferRuntimeException()
+    {
+        $exception = $this->readonlyConfig(
+            function () {
+                Config::setConfigData('foo', 'bar');
+            }
+        );
+
+        $this->assertInstanceOf(
+            'PHP_CodeSniffer\Exceptions\RuntimeException',
+            $exception,
+            'Expected exception class'
+        );
+
+    }//end testReadonlyConfigThrowsPhpCodeSnifferRuntimeException()
+
+
+    /**
+     * Test read-only config file throws a RuntimeException
+     *
+     * @return void
+     */
+    public function testReadonlyConfigSettingValue()
+    {
+        $exception = $this->readonlyConfig(
+            function () {
+                $args = array(
+                         '--config-set',
+                         'foo',
+                         'bar',
+                        );
+                new Config($args);
+            }
+        );
+
+        $this->assertNotNull(
+            $exception,
+            'An expected exception was not thrown'
+        );
+
+        $this->assertInstanceOf(
+            'PHP_CodeSniffer\Exceptions\DeepExitException',
+            $exception
+        );
+
+    }//end testReadonlyConfigSettingValue()
+
+
+    /**
+     * Make callback operate on a read-only config file, helper function
+     * for tests.
+     *
+     * @param callable $callback to be run with a readonly config
+     *
+     * @return NULL|\Exception
+     */
+    private function readonlyConfig($callback)
+    {
+        if (true === defined('PHP_WINDOWS_VERSION_BUILD')) {
+            $this->markTestSkipped('Test not valid for windows');
+        }
+
+        $file = $this->getConfigPath();
+
+        if (false === file_exists($this->configFile)) {
+            touch($file);
+        }
+
+        $this->assertFileExists(
+            $this->configFile,
+            'precondition: config file exists'
+        );
+        $this->assertTrue(
+            is_writable($this->configFile),
+            'precondition: config file is writable'
+        );
+
+        $result = fileperms($this->configFile);
+
+        $savedPermission = ($result & 0777);
+
+        $this->assertInternalType(
+            'int',
+            $result,
+            'precondition: fileperms available'
+        );
+
+        $this->assertTrue(
+            chmod($this->configFile, 0444),
+            'precondition: read-only config file'
+        );
+
+        $exception = null;
+        try {
+            call_user_func($callback);
+        } catch (\Exception $exception) {
+            // Fall-through intended.
+        }
+
+        chmod($this->configFile, $savedPermission);
+
+        return $exception;
+
+    }//end readonlyConfig()
+
+
+    /**
+     * Get path of the configuration file.
+     *
+     * @return string
+     */
+    private function getConfigPath()
+    {
+        return __DIR__.'/../../CodeSniffer.conf';
```

Comment:
It's in the `/data/PHP_CodeSniffer` directory in a PEAR install.

Typical directory layout for the relevant directories:
```
- PHP
     - /data
          - /PHP_CodeSniffer       <- this is where the `.conf` file will be
     - /PEAR
          - /PHP
               - /CodeSniffer      <- this is the PHP_CodeSniffer root
     - /tests
          - /PHP_CodeSniffer      <- this is where the tests live
```

It might be good to get a second opinion on this though as I have to admit this is based PHPCS 2.x as I haven't updated my PEAR install for a while - I'm using a stand-alone git checkout instead myself.